docs/guides/server/containers.adoc

```diff
@@ -251,4 +252,8 @@ podman|docker run --name mykeycloak -p 8080:8080 \
         start-dev
 ----
 
+WARNING: As the heap size is dynamically calculated based on the total container memory, you should *always set the memory limit* for the container.
```

Comment:
Is there a note somewhere that users can have the legacy behavior if use KC_RUN_IN_CONTAINER=false

Reply:
I rather directed users to set different memory limits [1]. I think it should be sufficient, right? The env property should be considered an internal one, as users might not need to touch it at all. But I'm open to suggestions. 

This is the backported version of the fix, so please add comments here: https://github.com/keycloak/keycloak/pull/28808

[1] https://github.com/keycloak/keycloak/pull/28808/files#diff-94b924a0a3ec356afd60a9253c3c2f192732e146877e83cdad98ce3b51926facR242